### PR TITLE
Move system tests functionality to tester, and reuse in staging tests

### PR DESCRIPTION
**Pull request by jonmv**
@kkraune please review.

### CODE REVIEW

**Comment by jonmv** [comment]
Basically the same tests, but moved code around a bit, so I could reuse it for staging tests. I also wondered if I should use the feed client there, but without a checked-in, ready feed it didn't make much sense to me.

File: src/test/java/ai/vespa/cloud/docsearch/VespaDocStagingSetup.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    public void feedAndVerify(TestReporter report) throws Exception {
+        VespaDocTester tester = new VespaDocTester(endpoint, report);
+        tester.removeAllTestDocs();
+        tester.feedTestDocs();
+
+        Set<DocumentId> ids = tester.getTestDocIDs();
+        for (DocumentId id : ids) report.publishEntry(id.toString());
+        assertEquals(5, ids.size(), "test-documents.json should have 5 documents");
+
+        String allQuery = "select * from sources * where sddocname contains \"doc\";";
```
Comment (kkraune) [inline src/test/java/ai/vespa/cloud/docsearch/VespaDocStagingSetup.java:31]: ; not needed anymore
```suggestion
        String allQuery = "select * from sources * where sddocname contains \"doc\"";
```

File: src/test/java/ai/vespa/cloud/docsearch/VespaDocProductionTest.java [hunk 2]
```diff
@@ -11,30 +11,28 @@
...
-
         // Can use publicly available resources only, so use open query interface
         // Here, ensure > 50 documents about ranking
         HttpRequest req = HttpRequest.newBuilder()
-                .GET()
-                .uri(URI.create("https://doc-search.vespa.oath.cloud/search/?query=ranking&ranking=documentation&locale=en-US&hits=1"))
-                .build();
-        HttpResponse<String> res = httpClient.send(req, HttpResponse.BodyHandlers.ofString());
+                                     .GET()
+                                     .uri(URI.create("https://doc-search.vespa.oath.cloud/search/?query=ranking&ranking=documentation&locale=en-US&hits=1"))
```
Comment (kkraune) [inline src/test/java/ai/vespa/cloud/docsearch/VespaDocProductionTest.java:25]: as this is a sample app, an YQL query would be better, but this does not change it for the worse :-)

File: src/test/java/ai/vespa/cloud/docsearch/VespaDocStagingSetup.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        tester.verifyQueryResults(ids, allQuery, "5s"); // Use a high timeout for first query
+
+        String accessQuery = "select * from sources * where content contains \"access\";";
```
Comment (kkraune) [inline src/test/java/ai/vespa/cloud/docsearch/VespaDocStagingSetup.java:34]:
```suggestion
        String accessQuery = "select * from sources * where content contains \"access\"";
```

File: src/test/java/ai/vespa/cloud/docsearch/VespaDocStagingTest.java [hunk 4]
```diff
@@ -1,14 +1,52 @@
...
-    public void testSearchAndFeeding() throws Exception {
+    @DisplayName("Verify documents, and further feeding, after upgrade")
+    public void verifySearchThenUpdateAndRemove(TestReporter report) throws Exception {
+        VespaDocTester tester = new VespaDocTester(endpoint, report);
+
+        Set<DocumentId> ids = tester.getTestDocIDs();
+        for (DocumentId id : ids) report.publishEntry(id.toString());
+        assertEquals(5, ids.size(), "test-documents.json should have 5 documents");
+
+        String allQuery = "select * from sources * where sddocname contains \"doc\";";
```
Comment (kkraune) [inline src/test/java/ai/vespa/cloud/docsearch/VespaDocStagingTest.java:29]:
```suggestion
        String allQuery = "select * from sources * where sddocname contains \"doc\"";
```

File: src/test/java/ai/vespa/cloud/docsearch/VespaDocStagingTest.java [hunk 5]
```diff
@@ -1,14 +1,52 @@
...
+        tester.verifyQueryResults(ids, allQuery, "5s"); // Use a high timeout for first query
+
+        String accessQuery = "select * from sources * where content contains \"access\";";
```
Comment (kkraune) [inline src/test/java/ai/vespa/cloud/docsearch/VespaDocStagingTest.java:32]:
```suggestion
        String accessQuery = "select * from sources * where content contains \"access\"";
```

File: src/test/java/ai/vespa/cloud/docsearch/VespaDocSystemTest.java [hunk 6]
```diff
@@ -1,228 +1,90 @@
...
+        tester.verifyQueryResults(ids, allQuery, "5s"); // Use a high timeout for first query
 
-    @Test
-    public void testUpdate() throws Exception {
-        updateTestDocs("/test-documents-updates.json");
-
-        HashSet<String> ids = getTestDocIDs();
-        ids.forEach(id -> {System.out.println(id);});
-        assertEquals(5, ids.size(), "test-documents-updates.json has 5 documents");
+        String accessQuery = "select * from sources * where content contains \"access\";";
```
Comment (kkraune) [inline src/test/java/ai/vespa/cloud/docsearch/VespaDocSystemTest.java:34]:
```suggestion
        String accessQuery = "select * from sources * where content contains \"access\"";
```

**Review by kkraune — APPROVED** [review]
Great!

File: src/test/java/ai/vespa/cloud/docsearch/VespaDocProductionTest.java (hunk 2 above)
Comment (jonmv) [reply src/test/java/ai/vespa/cloud/docsearch/VespaDocProductionTest.java:25]: Hehe. We have YQL in the other tests, though. Prodduction tests must be ddifferent, at least until we can provide them with read access.

**Comment by jonmv** [comment]
Thanks for fixes.

**Pull request merged**